azure-fs-aks/Program.fs

```diff
@@ -0,0 +1,131 @@
+
+    let createServicePrincipal name (app: Application) =
+        ServicePrincipal(name,
+            ServicePrincipalArgs(ApplicationId = io app.ApplicationId))
+
+    let createServicePrincipalPassword name (password: RandomPassword) (servicePrincipal: ServicePrincipal) =
+        ServicePrincipalPassword(name,
+            ServicePrincipalPasswordArgs(
+                ServicePrincipalId = io servicePrincipal.Id,
+                Value = io password.Result,
+                EndDate = input "2099-01-01T00:00:00Z"
+            ))
+
+    let createVnet name (resourceGroup: ResourceGroup) =
+        VirtualNetwork(name,
+            VirtualNetworkArgs(
+                ResourceGroupName = io resourceGroup.Name,
+                AddressSpaces = inputList [ input "10.2.0.0/16" ]
+            ))
+
+    let createSubnet name (vnet: VirtualNetwork) (resourceGroup: ResourceGroup) =
+        Subnet(name, 
+            SubnetArgs(
+                ResourceGroupName = io resourceGroup.Name,
+                VirtualNetworkName = io vnet.Name,
+                AddressPrefix = input "10.2.1.0/24"
+            ))
+
+    let createCluster name (subnet: Subnet) (privateKey: PrivateKey) (app: Application) (servicePrincipalPassword: ServicePrincipalPassword) (resourceGroup: ResourceGroup) nodeCount =
+        let defaultNodePoolArgs =
+            KubernetesClusterDefaultNodePoolArgs(
+                Name = input "aksagentpool",
+                NodeCount = input nodeCount,
+                VmSize = input "Standard_B2s",
+                OsDiskSizeGb = input 30,
+                VnetSubnetId = io subnet.Id
+            )
+
+        let linuxProfileArgs = 
+            let keyArgs = KubernetesClusterLinuxProfileSshKeyArgs(KeyData = io privateKey.PublicKeyOpenssh)
+            KubernetesClusterLinuxProfileArgs(
+                AdminUsername = input "aksuser",
+                SshKey = input keyArgs
+            )
+
+        let servicePrincipalArgs =
+            KubernetesClusterServicePrincipalArgs(
+                ClientId = io app.ApplicationId,
+                ClientSecret = io servicePrincipalPassword.Value
+            )
+
+        let rbacArgs =
+            KubernetesClusterRoleBasedAccessControlArgs(Enabled = input true)
+
+        let networkProfileArgs =
+            KubernetesClusterNetworkProfileArgs(
+                NetworkPlugin = input "azure",
+                DnsServiceIp = input "10.2.2.254",
+                ServiceCidr = input "10.2.2.0/24",
+                DockerBridgeCidr = input "172.17.0.1/16"
+            )
+
+        KubernetesCluster(name,
+            KubernetesClusterArgs(
+                ResourceGroupName = io resourceGroup.Name,
+                DefaultNodePool = input defaultNodePoolArgs,
+                DnsPrefix = input "fsaks",
+                LinuxProfile = input linuxProfileArgs,
+                ServicePrincipal = input servicePrincipalArgs,
+                KubernetesVersion = input "1.16.4",
```

Comment:
Make `version` a parameter?

Reply:
I was trying to have sane defaults in the helpers, but you might be right that it make sense to have the version as a parameter.

Reply:
That's the only thing I have to change **every single time** I run one of those AKS examples. They throw old versions away pretty fast.